docs/guide/dev/integration/react/v2/README.md

```diff
@@ -0,0 +1,526 @@
+---
+category: react-integration
+order: 40
+url: guide/dev_react_v2
+menu-title: v2
+meta-title-short: v2
+---
+
+<!--
+Copyright (c) 2003-2021, CKSource - Frederico Knabben. All rights reserved.
+For licensing, see LICENSE.md.
+-->
+
+# CKEditor 4 React Integration - v2
+
+<info-box info="">
+	This feature is provided through the <a href="https://www.npmjs.com/package/ckeditor4-react"><code>ckeditor4-react</code> npm package</a>.
+</info-box>
+
+CKEditor 4 offers a React integration through library described below. By providing a deep integration of CKEditor 4 and React we let you use the native features of the WYSIWYG editor inside your React app. This package is compatible with React version 16.8.6 or higher.
+
+## Basic Usage
+
+In order to create an editor instance in React, install the `ckeditor4-react` npm package as a dependency of your project:
+
+```plaintext
+npm install ckeditor4-react
+```
+
+Note that `react` is a peer dependency and `ckeditor4` is an **optional** peer dependency.
+
+This package exposes a high-level `CKEditor` component and a low-level `useCKEditor` hook. For simpler use cases `CKEditor` component should be a go-to solution. For advanced cases consider using `useCKEditor` hook.
+
+An example React app featuring `CKEditor` component would look like the following:
+
+```js
+import React from 'react';
+import { CKEditor } from 'ckeditor4-react';
+
+function App() {
+	return (
+		<div className="App">
+			<h2>Using CKEditor 4 in React</h2>
+			<CKEditor
+				initData={<p>Hello from CKEditor 4!</p>}
+				onInstanceReady={ () => {
+					alert( 'Editor is ready!' );
+				} }
+			/>
+		</div>
+	);
+}
+
+export default App;
+```
+
+## Customizing CKEditor Preset or Version
+
+By default, the CKEditor 4 React integration loads the [standard-all preset](https://ckeditor.com/cke4/presets-all) of the latest CKEditor 4 release from the [CDN](https://cdn.ckeditor.com/) when creating the first editor. This behavior can be altered by using the `editorUrl` prop to point to the desired CKEditor script location:
+
+```js
+// Hook
+const { editor } = useCKEditor( {
+	...props,
+	editorUrl: 'https://your-website.example/ckeditor/ckeditor.js'
+} );
+
+// Component
+const component = (
+	<CKEditor
+		{...props}
+		editorUrl="https://your-website.example/ckeditor/ckeditor.js"
+	/>
+);
+```
+
+Value of `editorUrl` must remain constant across all instances of editor.
+
+## CKEditor component
+
+The `CKEditor` component is a wrapper around low-level `useCKEditor` hook.
+
+### API
+
+The component's API exposes initial configuration options such as `config`, `editorUrl`, or `initData` as well as editor's event handlers. If you find `CKEditor` component's capabilities insufficient, consider using more flexible `useCKEditor` hook.
```

Comment:
```suggestion
The component's API exposes initial configuration options such as `config`, `editorUrl` or `initData` as well as editor's event handlers. If you find `CKEditor` component's capabilities insufficient, consider using a more flexible `useCKEditor` hook.
```

Reply:
Similarly to `and` rule, putting comma before `or` is [correct](https://www.grammarly.com/blog/comma-before-or/). I will use it consistently.